flink-clients/src/main/java/org/apache/flink/client/deployment/JobSubmitter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.client.deployment;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * An interface to be implemented by entities that are
+ * allowed/capable to submit {@link JobGraph JobGraphs} for execution.
+ */
+@Internal
+public interface JobSubmitter {
```

Comment:
The only usage of this interface is

```java
public class EmbeddedClient implements JobClient, JobSubmitter {
```

If we don't need this abstraction anywhere else, we should remove it; if we need it, I'll doubt why `ClusterClient` doesn't `extends` it.